admissions/utils.py

```diff
@@ -832,3 +897,59 @@ def get_applicant_offered_position(applicant):
         )
 
     return interest.position_to_be_offered
+
+
+def get_applicant_priority_list(applicant_id, priority_order):
+    trimmed_global_ids = []
+    for priority_order_id in priority_order:
+        if not priority_order_id:
+            continue
+        trimmed_global_ids.append(priority_order_id)
+
+    applicant_id = disambiguate_id(applicant_id)
+    applicant = Applicant.objects.get(id=applicant_id)
+    ids = [disambiguate_id(global_id) for global_id in trimmed_global_ids]
+
+    parsed_priorities = []
+    for id in ids:
+        parsed_priorities.append(InternalGroupPosition.objects.get(id=id))
+
+    return applicant, parsed_priorities
+
+
+def construct_new_priority_list(priority_order):
+    priority_order = [disambiguate_id(global_id) for global_id in priority_order]
+    return [InternalGroupPosition.objects.get(id=id) for id in priority_order]
```

Comment:
We can make use of the `__in` filter keyword which checks if a value is included in a list
```suggestion
    priority_order_ids = [disambiguate_id(global_id) for global_id in priority_order]
    return InternalGroupPosition.objects.filter(pk__in=priority_order_ids) 
```

Reply:
Looks like that doesn't preserve order, did not test it a lot